### ISSUE TITLE
IIVsearch tool uses wrong base model

### ISSUE THREAD

**Issue by liuxl17** [pharmpy/pharmpy#1017]
When I tested the same script, the recently released 0.74.1 version gave different iivsearch results from the previous 0.73.1 version.

Below is the script I used:
```
mfl <- "ABSORPTION([ZO,SEQ-ZO-FO]);LAGTIME();TRANSITS([1,3,10],*);PERIPHERALS(1);ELIMINATION([MM,MIX-FO-MM])"
order <- c('structural', 'iivsearch', 'residual', 'allometry', 'covariates')
res <- run_amd(input=model,
               modeltype='pk_oral',
               cl_init = 2700,
               vc_init = 10000,
               mat_init = 0.15,
               order=order,
               categorical=list('SEX'),
               continuous = list('AGE','WT'),
               allometric_variable='WT')
```
The two versions selected the same structural model which is `ABSORPTION(SEQ-ZO-FO);PERIPHERALS(1)`, which should be used as the base model in the subsequent step `iivsearch`. 
```
$PROBLEM ABSORPTION(SEQ-ZO-FO);PERIPHERALS(1)
$INPUT ID OID TIME DV AGE SEX HT WT DOSE AMT EVID MDV STUD NDV RATE
$SUBROUTINES ADVAN4  TRANS4 
$PK
VP1 = THETA(6)
QP1 = THETA(5)
MDT = THETA(4)*EXP(ETA(4))
VC = THETA(1) * EXP(ETA(1))
CL = THETA(2) * EXP(ETA(2))
MAT = THETA(3) * EXP(ETA(3))
KA = 1/MAT
V2 = VC
D1 = 2*MDT
Q = QP1
V3 = VP1
```
While version 0.74.1 used a different model as input. Below is the base model used in the two versions. 
base model (0.73.1)
```
$PROBLEM [VC,CL,MAT,MDT,QP1,VP1]
$INPUT ID OID TIME DV AGE SEX HT WT DOSE AMT EVID MDV STUD NDV RATE
$SUBROUTINES ADVAN4  TRANS4 
$PK
VP1 = THETA(6)*EXP(ETA(6))
QP1 = THETA(5)*EXP(ETA(5))
MDT = THETA(4)*EXP(ETA(4))
VC = THETA(1) * EXP(ETA(1))
CL = THETA(2) * EXP(ETA(2))
MAT = THETA(3) * EXP(ETA(3))
KA = 1/MAT
V2 = VC
D1 = 2*MDT
Q = QP1
V3 = VP1
```
base model (0.74.1)
```
$PROBLEM [VC,CL,MAT,MDT]
$INPUT ID OID TIME DV AGE SEX HT WT DOSE AMT EVID MDV STUD NDV RATE
$SUBROUTINES ADVAN2  TRANS2 
$PK
MDT = THETA(4)*EXP(ETA(4))
VC = THETA(1) * EXP(ETA(1))
CL = THETA(2) * EXP(ETA(2))
MAT = THETA(3) * EXP(ETA(3))
KA = 1/MAT
V = VC
D1 = 2*MDT
```
As a result, they gave different IIVsearch results. The best IIV model of 0.73.1 is `[MAT]+[VC,MDT,QP1]` while 0.74.1 is `[VC,CL,MDT]`.

This may have something to do with the disordered ranking shown in the `result.csv` file of `modelsearch` tool. Below shows the `result.csv` file of version 0.74.1. The "rank" column is not consistent with the actual ranking as how they are listed (which is sorted by dbic).

![image](https://user-images.githubusercontent.com/53524647/182572782-d504b92b-09f4-4dc4-9fba-3d5297db71eb.png)

Then I compared the model file of modelsearch_candidate19 (highest dbic) and modelsearch_candidate2 (the rank is 1). 
modelsearch_candidate19 (with highest dbic) 
```
$PROBLEM ABSORPTION(SEQ-ZO-FO);PERIPHERALS(1)
$INPUT ID OID TIME DV AGE SEX HT WT DOSE AMT EVID MDV STUD NDV RATE
$SUBROUTINES ADVAN4  TRANS4 
$PK
VP1 = THETA(6)
QP1 = THETA(5)
MDT = THETA(4)*EXP(ETA(4))
VC = THETA(1) * EXP(ETA(1))
CL = THETA(2) * EXP(ETA(2))
MAT = THETA(3) * EXP(ETA(3))
KA = 1/MAT
V2 = VC
D1 = 2*MDT
Q = QP1
V3 = VP1
```
modelsearch_candidate2 (the rank is 1)
```
$PROBLEM ABSORPTION(SEQ-ZO-FO)
$INPUT ID OID TIME DV AGE SEX HT WT DOSE AMT EVID MDV STUD NDV RATE
$SUBROUTINES ADVAN2  TRANS2 
$PK
MDT = THETA(4)*EXP(ETA(4))
VC = THETA(1) * EXP(ETA(1))
CL = THETA(2) * EXP(ETA(2))
MAT = THETA(3) * EXP(ETA(3))
KA = 1/MAT
V = VC
D1 = 2*MDT
```
Apparently, modelsearch_candidate19 is the best model in the `modelsearch` step but in version 0.74.1, modelsearch_candidate2 was selected as the base model used in the `iivsearch` step.

**Comment by stellabelin** [comment]
Hi! Thank you for reporting this, I will look into the ranking directly.

**Comment by stellabelin** [comment]
Hi! I could recreate the issue in 0.74.1. I tested in the development version and it looks correct, this was fixed in 26ff49b83cc486eafa47dd6b89190130743e0e98 and will be in the next release. I'm closing the issue, please reopen if you still gets problem in the next release!
